backend/amd64/arch.ml

```diff
@@ -15,99 +15,84 @@
 (**************************************************************************)
 [@@@ocaml.warning "+4"]
 
-(* POPCNT instruction is not available prior to Nehalem, released in 2008. *)
-let popcnt_support = ref true
-
-(* PREFETCHW instruction is not available on processors
-   based on Haswell or earlier microarchitectures. *)
-let prefetchw_support = ref true
-
-(* PREFETCHWT1 is Intel Xeon Phi only. *)
-let prefetchwt1_support = ref false
+module Extension = struct
+  module T = struct
+    type t =
+      | POPCNT
+      | PREFETCHW
+      | PREFETCHWT1
+      | SSE3
+      | SSSE3
+      | SSE4_1
+      | SSE4_2
+      | CLMUL
+      | BMI
+      | BMI2
+
+    let compare = compare
+  end
+
+  include T
+  module Set = Set.Make(T)
+
+  let all = [ POPCNT; PREFETCHW; PREFETCHWT1; SSE3; SSSE3; SSE4_1; SSE4_2; CLMUL; BMI; BMI2 ]
+
+  let config = ref (Set.of_list [ POPCNT; PREFETCHW; SSE3; SSSE3; SSE4_1; SSE4_2; CLMUL; BMI; BMI2 ])
```

Comment:
A new flag would be false by default, unless it is explicitly added to "config". This is a nitpick but it may be nicer to have a function `default : t -> bool` and filter `all` on it, so the initial status of the flag is easy to see.

Reply:
Sure, added